test_unstructured/partition/utils/ocr_models/test_ocr_interface.py

```diff
@@ -39,18 +41,19 @@ def it_provides_access_to_the_configured_OCR_agent(
         get_instance_.assert_called_once_with(OCR_AGENT_TESSERACT)
         assert ocr_agent is ocr_agent_
 
-    @pytest.mark.parametrize("ExceptionCls", [ImportError, AttributeError])
-    def but_it_raises_whan_no_such_ocr_agent_class_is_found(
-        self, ExceptionCls: type, _get_ocr_agent_cls_qname_: Mock, get_instance_: Mock
-    ):
+    def but_it_raises_when_no_such_ocr_agent_class_is_found(self, _get_ocr_agent_cls_qname_: Mock):
         _get_ocr_agent_cls_qname_.return_value = "Invalid.Ocr.Agent.Qname"
-        get_instance_.side_effect = ExceptionCls
-
-        with pytest.raises(ValueError, match="OCR_AGENT must be set to an existing OCR agent "):
+        with pytest.raises(ValueError) as exc_info:
             OCRAgent.get_agent()
+        assert "must be set to a whitelisted module" in str(exc_info.value)
 
-        _get_ocr_agent_cls_qname_.assert_called_once_with()
-        get_instance_.assert_called_once_with("Invalid.Ocr.Agent.Qname")
+        _get_ocr_agent_cls_qname_.return_value = OCR_AGENT_TESSERACT
+        for exception_cls in [ImportError, AttributeError]:
+            with patch("importlib.import_module", side_effect=exception_cls), pytest.raises(
+                RuntimeError
+            ) as exc_info:
+                OCRAgent.get_agent()
+            assert "Could not get the OCRAgent instance" in str(exc_info.value)
```

Comment:
I would consider these two distinct tests, each for a different behavior. I'd suggest breaking them apart like this:
```python
    def but_it_raises_when_the_requested_agent_is_not_whitelisted(
        self, _get_ocr_agent_cls_qname_: Mock
    ):
        _get_ocr_agent_cls_qname_.return_value = "Invalid.Ocr.Agent.Qname"
        with pytest.raises(ValueError, match="must be set to a whitelisted module"):
            OCRAgent.get_agent()

    @pytest.mark.parametrize("exception_cls", [ImportError, AttributeError])
    def and_it_raises_when_the_requested_agent_cannot_be_loaded(
        self, _get_ocr_agent_cls_qname_: Mock, exception_cls: type[Exception]
    ):
        _get_ocr_agent_cls_qname_.return_value = OCR_AGENT_TESSERACT
        with patch("importlib.import_module", side_effect=exception_cls), pytest.raises(
            RuntimeError, match="Could not get the OCRAgent instance"
        ):
            OCRAgent.get_agent()
```

One fine point in the change other than splitting the two, testing the exception message (always a good idea I think) is easy to do with the `match="..."` arg of `pytest.raises()`. Saves adding an `as exc_info` clause and the need for an extra assert. Also keeps the exception detail all on one line so it's easy to grasp at a glance.

The `match` arg is a regex. I rarely find the need for that but good to know in case you want to use anchors or placeholders :)